src/Components/ArtistFilter/Artworks.tsx

```diff
@@ -14,40 +9,70 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
 
 interface Props {
   filtered_artworks: Artworks_filtered_artworks
-  relay: RelayPaginationProp
+  relay: RelayRefetchProp
   artistID: string
 }
 
 const PAGE_SIZE = 10
 
 class Artworks extends React.Component<Props> {
-  loadMoreArtworks(filters) {
-    const hasMore = this.props.filtered_artworks.artworks.pageInfo.hasNextPage
-    const endCursor = this.props.filtered_artworks.artworks.pageInfo.endCursor
-    if (hasMore) {
-      // TODO: Should refetchConnection keep appending records?
-      this.props.relay.refetchConnection(
-        PAGE_SIZE,
-        error => {
-          if (error) {
-            console.error(error)
-          }
-          filters.incrementPage()
-        },
-        { cursor: endCursor }
+  loadArtworks(filters, cursor, page) {
+    this.props.relay.refetch(
+      {
+        first: PAGE_SIZE,
+        after: cursor,
+        filteredArtworksNodeID: this.props.filtered_artworks.__id,
+      },
+      null,
+      error => {
+        if (error) {
+          console.error(error)
+        }
+        filters.setPage(page)
+      }
+    )
+  }
+  renderPages(filters, pageCursors, options = { prefix: "", suffix: null }) {
+    return pageCursors.map(pageCursor => {
+      const { cursor, isCurrent, page } = pageCursor
+      const { prefix, suffix } = options
+      return (
+        <span
+          onClick={() => {
+            this.loadArtworks(filters, cursor, page)
+          }}
+          style={{
+            padding: `${isCurrent ? "0 10px 0 10px" : "0 3px 0 0"}`,
+          }}
+        >
+          {prefix ? prefix + page : page}
+          {suffix}
+        </span>
       )
-    }
+    })
   }
   renderPagination(filters) {
+    const {
+      first,
+      last,
+      around,
+    } = this.props.filtered_artworks.artworks.pageCursors
     return (
       <div>
-        <div>Current Page: {filters.state.page}</div>
-        <div
-          onClick={() => {
-            this.loadMoreArtworks(filters)
-          }}
-        >
-          Next
+        <div>
+          {first
```

Comment:
This is the only 'display' logic in terms of, any sort of separator in the pagination bar.